jarviscli/plugins/typingtest.py

```diff
@@ -0,0 +1,199 @@
+    clean_text = clean_text.replace('.', '. ')
+    return clean_text
+
+
+def format_time(seconds):
+    new_minutes = int(seconds/60)
+    new_seconds = int(seconds - new_minutes * 60)
+
+    new_minutes = str(new_minutes)
+    new_seconds = str(new_seconds)
+
+    new_minutes = '0'*(2-len(new_minutes)) + new_minutes
+    new_seconds = '0'*(2-len(new_seconds)) + new_seconds
+
+    return new_minutes + ':' + new_seconds
+
+
+def take_input():
+    entered_letter = getch()
+    if (entered_letter == '='):
+        game_end()
+    letter_object = info['letters'][info['position']]
+    print(letter_object.letter)
+    if (entered_letter == letter_object.letter):
+        letter_object.state = Letter.State.CORRECT
+        info['score'] += 1
+        if(entered_letter == ' '):
+            if(not info['mistake_in_current_word']):
+                info['words'] += 1
+            info['mistake_in_current_word'] = False
+    else:
+        letter_object.state = Letter.State.INCORRECT
+        info['mistake_in_current_word'] = True
+    info['position'] += 1
+
+
+def reset():
+    info['state'] = GameState.RUNNING
+    info['text'] = get_text()
+    info['words'] = 0
+    info['time_left'] = total_time
+    info['letters'] = ['-1'] * len(info['text'])
+    info['score'] = 0
+    info['position'] = 0
+    info['mistake_in_current_word'] = False
+
+
+def game_start():
+    reset()
+    for i in range(len(info['text'])):
+        print(info['text'][i])
+        info['letters'][i] = Letter()
+        info['letters'][i].setLetter(info['text'][i])
+
+    game_running()
+
+
+def game_running():
+    while(info['state'] == GameState.RUNNING):
+        time_0 = time.time()
+
+        print_screen()
+        take_input()
+
+        time_1 = time.time()
+        info['time_left'] -= (time_1 - time_0)
+        if(info['time_left'] <= 0 or info['position'] == len(info['text'])):
+            game_end()
+
+
+def game_end():
+    info['state'] = GameState.ENDED
+    wpm = info['words'] / ((total_time - info['time_left']) / 60)
+    print(default_text_color+'\n\nWords per minute - {}'.format(wpm))
+    if(wpm >= 20):
+        with open(os.path.join(FILE_PATH, "../data/typing_test_data.csv"),
+                  mode='r') as file:
+            data = list(csv.reader(file))
+            for i in range(len(data)):
+                if int(data[i][0]) > int(wpm) and i != 0:
+                    print('You are better than {}% of people!'
+                          .format(float(data[i-1][1]) * 100))
+                    break
+
+
+def print_screen():
+    print("\033c")
+    string = 'Welcome to the typing text game. Enter the letters shown \
+        below (\'=\' to quit) - \n'
+    string += '\nScore - {}\t Time - {}\t Words -  \
+        {}\n\n'.format(info['score'], format_time(info['time_left']),
+                       info['words'])
+    for i in range(len(info['letters'])):
+        string += str(info['letters'][i])
+    sys.stdout.write(string)
```

Comment:
why are you using stdout.write in one place, print in another, when you have jarvis.say?